pennylane/qnode.py

```diff
@@ -605,8 +605,23 @@ def __call__(self, *args, **kwargs):
             # construct the tape
             self.construct(args, kwargs)
 
-        # execute the tape
-        res = self.qtape.execute(device=self.device)
+        # ************************
+        supports_hamiltonian = self.device.supports_observable("Hamiltonian")
+        hamiltonian_in_obs = "Hamiltonian" in [obs.name for obs in self.qtape.observables]
+
+        if hamiltonian_in_obs and (self.device.shots is not None or not supports_hamiltonian):
+            try:
+                tapes, fn = qml.transforms.hamiltonian_expand(self.qtape)
+            except ValueError:
+                raise ValueError("At the moment, only single expectations of Hamiltonian observables can be returned"
+                                 "on the {} device.".format(self.device.name))
+
+            results = [tape.execute(device=self.device) for tape in tapes]
+            res = fn(results)
+        else:
+            # this is what currently happens with the tape
+            res = self.qtape.execute(device=self.device)
+        # *************************

```

Comment:
Just added this to play around with this VQE implementation, it is not needed for the changes in the prototype

Reply:
Currently still faulty!